internal/oidc/oidc.go

```diff
@@ -29,9 +31,25 @@ func PinnipedCLIOIDCClient() *fosite.DefaultOpenIDConnectClient {
 	}
 }
 
-func FositeOauth2Helper(oauthStore interface{}, hmacSecretOfLengthAtLeast32 []byte) fosite.OAuth2Provider {
+func FositeOauth2Helper(issuerURL string, oauthStore fosite.Storage, hmacSecretOfLengthAtLeast32 []byte) fosite.OAuth2Provider {
 	oauthConfig := &compose.Config{
-		EnforcePKCEForPublicClients: true,
+		AuthorizeCodeLifespan: 3 * time.Minute, // seems more than long enough to exchange a code
+
+		IDTokenLifespan:     5 * time.Minute, // match clientCertificateTTL since it has similar properties to this token
+		AccessTokenLifespan: 5 * time.Minute, // match clientCertificateTTL since it has similar properties to this token
+
+		RefreshTokenLifespan: 16 * time.Hour, // long enough for a single workday
+
+		IDTokenIssuer: issuerURL,
+		TokenURL:      "", // TODO set once we have this endpoint written
+
+		ScopeStrategy:            fosite.ExactScopeStrategy, // be careful and only support exact string matching for scopes
+		AudienceMatchingStrategy: nil,                       // I believe the default is fine
+		EnforcePKCE:              true,                      // follow current set of best practices and always require PKCE
```

Comment:
We should require PKCE for public clients (`EnforcePKCEForPublicClients: true`) for sure. However, for an authcode flow with a private client, isn't PKCE overkill? We don't have any private clients yet. The only difference now is a slightly less specific error message gets returned when you don't use PKCE if you make `EnforcePKCE: true`.